common/txmgr/inmemory_store.go

```diff
@@ -79,11 +81,20 @@ func NewInMemoryStore[
 		ms.maxUnstarted = 10000
 	}
 
+	addressesToTxs := map[ADDR][]txmgrtypes.Tx[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE]{}
+	// populate all enabled addresses
+	enabledAddresses, err := keyStore.EnabledAddressesForChain(ctx, chainID)
```

Comment:
Wouldn't this just be an extra call to the DB if we build the address state out of the from addresses in tx's below? Don't think this would cause an issue though. There's a chance a tx doesn't exist for an enabled from address at startup but the code in `CreateTransaction` would take care of that. If this is causing problems with tests, maybe we should adjust those. If this is needed to get this through for now though, I'm ok with leaving it

Reply:
I'm good with that 👍